Gcpe.Hub.API/Controllers/UserPreferencesController.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using AutoMapper;
+using Gcpe.Hub.Data.Entity;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+namespace Gcpe.Hub.API.Controllers
+{
+    [Route("api/[Controller]")]
+    [ApiController]
+    [Produces("application/json")]
+    public class UserPreferencesController : BaseController
+    {
+        private readonly HubDbContext dbContext;
+        private readonly IMapper mapper;
+
+        public UserPreferencesController(HubDbContext dbContext, ILogger<UserPreferencesController> logger, IMapper mapper) : base(logger)
+        {
+            this.dbContext = dbContext;
+            this.mapper = mapper;
+        }
+
+        [HttpGet(Name = "GetUserMinistryPreferences")]
+        [Authorize(Policy = "ReadAccess")]
+        [Produces(typeof(string[]))]
+        [ProducesResponseType(400)]
+        [ProducesResponseType(404)]
+        [ResponseCache(Duration = 5)]
+        public IActionResult GetUserMinistryPreferences(bool getAbbreviations = false)
```

Comment:
It would be cleaner to just return a list of keys, since now we have a ministry service